sympy/integrals/transforms.py

```diff
@@ -1826,19 +1767,88 @@ def _collapse_extra(self, extra):
         return plane, cond
 
     def _try_directly(self, **hints):
-        fn = self.function
-        debug('----> _try_directly: %s'%(fn, ))
-        t_ = self.function_variable
-        s_ = self.transform_variable
-        LT = None
-        if not fn.is_Add:
-            fn = expand_mul(fn)
+        T = None
+        try_directly = not any(func.has(self.function_variable)
+                               for func in self.function.atoms(AppliedUndef))
+        if try_directly:
             try:
-                LT = self._compute_transform(fn, t_, s_, **hints)
+                T = self._compute_transform(self.function,
+                    self.function_variable, self.transform_variable, **hints)
             except IntegralTransformError:
-                LT = None
-        return fn, LT
+                T = None
 
+        return self.function, T
+
+    def doit(self, **hints):
+        """
+        Try to evaluate the transform in closed form.
+
+        Explanation
+        ===========
+
+        Standard hints are the following:
+        - ``noconds``:  if True, do not return convergence conditions. This is
+        the default behaviour.
+        """
+        _noconds = hints.get('noconds', False)
+        _simplify = hints.get('simplify', True)
+
+        debug('[LT doit] (%s, %s, %s)'%(self.function,
+                                        self.function_variable,
+                                        self.transform_variable))
+
+        t_ = self.function_variable
+        s_ = self.transform_variable
+        fn = self.function
+
+        terms = Add.make_args(fn)
+        results = []
+        for ff in terms:
+            k, f = ff.as_independent(t_, as_Add=False)
+            d, r = _laplace_apply_simple_rules(f, t_, s_)
+            if d:
+                results.append((k*r[0], r[1], r[2]))
+                continue
+            d, r = _laplace_apply_prog_rules(f, t_, s_)
+            if d:
+                results.append((k*r[0], r[1], r[2]))
+                continue
+            d, r = _laplace_expand(f, t_, s_)
+            if d:
+                results.append((k*r[0], r[1], r[2]))
+                continue
+            T = None
+            try_directly = not any(func.has(t_)
+                                    for func in f.atoms(AppliedUndef))
+            if try_directly:
```

Comment:
What exactly is the reason for this check?

Reply:
I don't know, it has always been in the `_try_directly` of `IntegralTransform`. All `test_laplace_transform` still run if I remove it, so I do. Change (M)

Reply:
I'm guessing it's because the integration is unlikely to succeed.